src/main/scala/util/ShiftReg.scala

```diff
@@ -84,11 +85,41 @@ object AsyncResetShiftReg {
     apply (in, depth, init.litValue.toInt, None)
 }
 
+class SynchronizerPrimitiveShiftRegVec(depth: Int = 1, init: Int = 0) extends AbstractPipelineReg(1) {
+  require(depth > 0, "Sync must be greater than 0.")
+  override def desiredName = s"SynchronizerPrimitiveShiftRegVec_d${depth}_i${init}"
+
+  val chain = List.tabulate(depth) { i =>
+    val reg = Module (new SynchronizerPrimitiveReg(init))
+    reg.io.clk := clock
+    reg.io.rst := reset
+    reg
+  }
+
+  chain.last.io.d := io.d.asBool
+  chain.last.io.en := Bool(true)
+
+  (chain.init zip chain.tail).foreach { case (sink, source) =>
+    sink.io.d := source.io.q
+    sink.io.en := Bool(true)
+  }
+  io.q := chain.head.io.q.asUInt
+}
+
+object SynchronizerPrimitiveShiftRegVec {
+  def apply [T <: Chisel.Data](in: T, depth: Int, init: Int): T =
+    AbstractPipelineReg(new SynchronizerPrimitiveShiftRegVec(depth, init), in)
+}
+
 // Note that it is important to override "name" in order to ensure that the Chisel dedup does
 // not try to merge instances of this with instances of the superclass.
-class AsyncResetSynchronizerShiftReg(w: Int = 1, sync: Int = 3, init: Int = 0) extends AsyncResetShiftReg(w, depth = sync, init, name = "sync") {
+class AsyncResetSynchronizerShiftReg(w: Int = 1, sync: Int = 3, init: Int = 0) extends AbstractPipelineReg(w) {
   require(sync > 0, "Sync must be greater than 0.")
   override def desiredName = s"AsyncResetSynchronizerShiftReg_w${w}_d${sync}_i${init}"
+
+  val output = Seq.tabulate(w) { i => SynchronizerPrimitiveShiftRegVec(io.d(i), depth = sync, init) }
```

Comment:
this looks good here... but I think we need to do something similar for the other `SynchronizerShiftReg` variants (or get rid of them)

Reply:
Sure, I changed `SyncResetSynchronizerShiftReg` to be constructed similarly. Basically WxD-sized reset synchronizers are created with W instantiations of D-deep 1-bit wide synchronizer shift regs. The other types are unchanged (constructed from D instances of a W-bit wide reg vec).